build.rs

```diff
@@ -11,24 +11,22 @@ fn main() {
         println!("cargo:rustc-link-lib=dylib=stdc++");
 
     } else {
-        let tokens: Vec<&str> = quadruple.split('-').collect();
-        // [0] is <arch><sub> | [1] is <vendor> | [2] is <sys> | [3] is <abi>
-        let sys = tokens[2];
-
-        if sys == "linux" {                         //=> assume compiling for Linux host
+        if CARGO_CFG_TARGET_OS == "linux" {                         //=> assume compiling for Linux host
```

Comment:
`CARGO_CFG_TARGET_OS` is an environment variable…

Reply:
This is not the _pushed too soon_ you're looking for.